modules/020-deckhouse/docs/internal/RELEASE.md

```diff
@@ -0,0 +1,23 @@
+# DeckhouseRelease details
+
+## Available annotations
+
+| annotation                               | description                                                                        |
+|------------------------------------------|------------------------------------------------------------------------------------|
+| release.deckhouse.io/force               | Apply specified release without any checks. Force deploy.                          |
+| release.deckhouse.io/disruption-approved | Approve release with disruptive changes. Works if update.disruptionMode is Manual. |
+| release.deckhouse.io/approved            | Approve release for deployment. Works if update.mode is Manual.                    |
+
+## Disruptive release
+
+It's a release with some potentially dangerous changes (change some default value / behavior / docker -> containerd, etc)
+To handle this release, you should add disruption check logic in a release `X-1`, for example - register requirements.DisruptionFunc in init() [example](modules/402-ingress-nginx/hooks/requirements.go)
+And then in a release `X` you should add record `"disruption:$functionName": "true"` to the [requirements.json](requirements.json) file.
+For example:
+In the release 1.35.0 set disruption check logic into ingressNginx
+In the release 1.36.0 set `"disruption:ingressNginx": "true"` to the requirements.json
```

Comment:
Imagine stable release channel. Can we add a disruption check in 1.35.5 (not 1.35.0), and update to 1.36.10 for this machinery to work?

Reply:
We could and this will work but we want to avoid this behavior. Because in a some airgap installation client can jump from 1.35.3 to 1.36.10. And 1.35.3 wouldn't have this machinery, so we will meet a problem. It's just an agreement, cause as I said, machinery will work here

Reply:
Ok. The development flow is clear to me:

To have a disruptive update in `1.X.*`, one should do:

1. Make a PR to `1.(X-1).0` with **both** precondition check and update requirement. In `requirements.json' the check should be turned on, but the update requirement is not.
2. Make PR  to `1.X.0` which turns the update requirement on in `requirements.json`.